messageflux/pydantic_handler.py

```diff
@@ -0,0 +1,57 @@
+import json
+from abc import abstractmethod, ABCMeta
+from dataclasses import dataclass
+from typing import Optional, TypeVar, Generic, Any
+
+try:
+    from pydantic import BaseModel, parse_raw_as
+except ImportError as ex:
+    raise ImportError('Please Install the required extra: messageflux[pydantic]') from ex
+
+from messageflux import InputDevice
+from messageflux.iodevices.base.common import MessageBundle, Message
+from messageflux.pipeline_service import PipelineHandlerBase, PipelineResult
+
+
+@dataclass
+class PydanticPipelineResult:
+    """
+    a result from pipeline handler
+    """
+    output_device_name: str
+    pydantic_object: Any
+
+
+T = TypeVar('T')
+
+
+class PydanticPipelineHandler(PipelineHandlerBase, Generic[T], metaclass=ABCMeta):
```

Comment:
Why do we need a generic here? If the developer will declare its own type in the `handle_object` function.

Reply:
Inspired by the fastapi approach, the message handling function itself can get validated on runtime only. And not on declare. In my pubsub implementation, I did it when the decorator is called.

Reply:
this will probably be the final solution

Reply:
I'll add a suggestion soon